core/lib/include/xfa.h

```diff
@@ -76,6 +77,7 @@ _Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
     _XFA_CONST(name, 0_) type name [0] = {}; \
     _XFA_CONST(name, 9_) type name ## _end [0] = {}; \
     _Pragma("GCC diagnostic pop") \
+    _Static_assert(__alignof__(type) >= ARCHITECTURE_WORD_BYTES, #type " is not aligned properly"); \
```

Comment:
I'm not sure about the `>=` here. Maybe `__alignof__(type) % ARCHITECTURE_WORD_BYTES == 0` is more correct.

I did some brief testing if an 8b aligned type still works with `>=` and it does.